packages/ethereum-contracts/contracts/libs/CFALibraryV1.sol

```diff
@@ -0,0 +1,285 @@
+//SPDX-License-Identifier: AGPLv3
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import {
+    ISuperfluid,
+    ISuperfluidToken
+} from "../interfaces/superfluid/ISuperfluid.sol";
+
+import {
+    IConstantFlowAgreementV1
+} from "../interfaces/agreements/IConstantFlowAgreementV1.sol";
+
+library CFALibraryV1 {
+
+    struct InitData {
+        ISuperfluid host;
+        IConstantFlowAgreementV1 cfa;
+    }
+
+    //@dev for working with the constant flow agreement within solidity
+    //the first set of functions are each for callAgreement()
+    //the second set of functions are each for use in callAgreementWithContext()
+
+     //create flow without userData
+    function createFlow(
+        InitData storage cfaLibrary,
+        address receiver,
+        ISuperfluidToken token,
+        int96 flowRate
+    ) internal {
+        cfaLibrary.host.callAgreement(
+            cfaLibrary.cfa,
+            abi.encodeWithSelector(
+                cfaLibrary.cfa.createFlow.selector,
+                token,
+                receiver,
+                flowRate,
+                new bytes(0) // placeholder
+            ),
+            "0x" //empty user data
+        );
+    }
+    
+
+    //create flow with userData
+    function createFlow(
+        InitData storage cfaLibrary, 
+        address receiver,
+        ISuperfluidToken token,
+        int96 flowRate,
+        bytes memory userData
+    ) internal {
+        cfaLibrary.host.callAgreement(
+            cfaLibrary.cfa,
+            abi.encodeWithSelector(
+                cfaLibrary.cfa.createFlow.selector,
+                token,
+                receiver,
+                flowRate,
+                new bytes(0) // placeholder
+            ),
+            userData
+        );
+    }
+
+    //update flow without userData
+    function updateFlow(
+        InitData storage cfaLibrary,
+        address receiver,
+        ISuperfluidToken token,
+        int96 flowRate
+    ) internal {
+        cfaLibrary.host.callAgreement(
+            cfaLibrary.cfa,
+            abi.encodeWithSelector(
+                cfaLibrary.cfa.updateFlow.selector,
+                token,
+                receiver,
+                flowRate,
+                new bytes(0) // placeholder
+            ),
+            "0x" //empty user data
+        );
+    }
+    
+
+    //update flow with userData
+    function updateFlow(
+        InitData memory cfaLibrary,
```

Comment:
"storage" should do, and the rest.

Reply:
Got it - will change 'memory' to 'storage' both here and elsewhere in the lib 👍🏻